### PR TITLE
WIP: AKS support

### PR DESCRIPTION
**Pull request by wbreza**
This is a very early POC of AKS support for `azd`.

### CODE REVIEW

File: templates/todo/projects/nodejs-mongo-aks/.repo/bicep/infra/main.bicep [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+@description('Primary location for all resources')
+param location string
+
+@description('Id of the user or app to assign application roles')
+param principalId string = ''
+
+@description('The AKS cluser username')
+param clusterAdminUsername string = 'aksadminuser'
+
+@secure()
```
Comment (ellismg) [inline templates/todo/projects/nodejs-mongo-aks/.repo/bicep/infra/main.bicep:18]: Why would the public key be marked with secure?

File: cli/azd/pkg/project/service_target_aks.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	config  *ServiceConfig
+	env     *environment.Environment
+	scope   *environment.DeploymentScope
+	az      azcli.AzCli
+	docker  docker.Docker
+	kubectl kubectl.KubectlCli
+}
+
+func (t *aksTarget) RequiredExternalTools() []tools.ExternalTool {
+	return []tools.ExternalTool{t.az, t.docker}
```
Comment (ellismg) [inline cli/azd/pkg/project/service_target_aks.go:28]: Should `t.kubectl` also be in this list?

File: cli/azd/pkg/project/service_target_aks.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	progress <- "Pushing container image"
+	if err := t.docker.Push(ctx, t.config.Path(), fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("pushing image: %w", err)
+	}
+
+	endpoints, err := t.Endpoints(ctx)
+	if err != nil {
+		return ServiceDeploymentResult{}, err
+	}
+
```
Comment (ellismg) [inline cli/azd/pkg/project/service_target_aks.go:83]: Guessing there will be a few more steps here that actualy end up doing some form of kubectl apply?

File: cli/azd/pkg/project/service_target_aks.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		TargetResourceId: azure.ContainerAppRID(t.env.GetSubscriptionId(), t.scope.ResourceGroupName(), t.scope.ResourceName()),
+		Kind:             ContainerAppTarget,
+		Details:          nil,
+		Endpoints:        endpoints,
+	}, nil
+}
+
+func (t *aksTarget) Endpoints(ctx context.Context) ([]string, error) {
+	// TODO Update
+	return []string{"https://aks.azure.com/sample"}, nil
```
Comment (ellismg) [inline cli/azd/pkg/project/service_target_aks.go:94]: FWIW - returning nil, nil here is a reasonable implementation (it just means we won't show an endpoint in the deploy output)

**Review by ellismg — COMMENTED** [review]
Interested in seeing what the plan for deploying services into the cluster ends up being, and also how you specify which cluster (and probably some k8s specific target stuff as well) for a given service.  Are we going to stick with a tagging strategy?

File: cli/azd/pkg/project/service_target_aks.go (hunk 3 above)
Comment (wbreza) [reply cli/azd/pkg/project/service_target_aks.go:83]: yeah - this is very early, just wanted to get something out there for any of you curious to see the journey.

**Pull request closed without merging**